File: frontend/src/api/pnl.ts
import { apiClient } from "./client";
import { PnlSummary } from "../types/domain";

export async function fetchPnlSummary() {
  const { data } = await apiClient.get<{ ok: boolean; data: PnlSummary }>("/api/pnl/summary");
  return data;
}

export async function fetchPnlHistory(limit = 60) {
  const { data } = await apiClient.get<{ ok: boolean; count: number; items: Array<Record<string, unknown>> }>("/api/pnl/history", {
    params: { limit }
  });
  return data;
}

export async function fetchPnlByStrategy() {
  const { data } = await apiClient.get<{ ok: boolean; count: number; items: Array<Record<string, unknown>> }>("/api/pnl/by-strategy");
  return data;
}

File: backend/app/api/v1/pnl.py
from __future__ import annotations

from fastapi import APIRouter, Depends, Query

from app.api.deps import get_pnl_service
from app.services.pnl_service import PnlService

router = APIRouter(prefix="/api/pnl", tags=["pnl"])


@router.get("/summary")
def pnl_summary(service: PnlService = Depends(get_pnl_service)) -> dict:
    return {"ok": True, "data": service.summary()}


@router.get("/history")
def pnl_history(
    limit: int = Query(default=120, ge=1, le=1000),
    service: PnlService = Depends(get_pnl_service),
) -> dict:
    items = service.history(limit=limit)
    return {"ok": True, "count": len(items), "items": items}


@router.get("/by-strategy")
def pnl_by_strategy(service: PnlService = Depends(get_pnl_service)) -> dict:
    items = service.by_strategy()
    return {"ok": True, "count": len(items), "items": items}
